File: C#api/Tests/test_GET_handler.cs
using System.Net.Http;
using System.Net;
using System.Threading.Tasks;
using Newtonsoft.Json;
using Models;
using Xunit;

public class ApiGetTests
{
    private readonly HttpClient _client;

    public ApiGetTests()
    {
        _client = new HttpClient { BaseAddress = new Uri("http://localhost:3000/api/v1/") };
        _client.DefaultRequestHeaders.Add("API_KEY", "a1b2c3d4e5");
    }
    #region Clients
    [Fact]
    public async Task Get_All_Clients()
    {
        var response = await _client.GetAsync("clients");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualClients = JsonConvert.DeserializeObject<List<Client>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_client_pool().GetClients(), actualClients);
    }

    [Fact]
    public async Task Get_Client_By_Id()
    {
        var clientId = 1; // Assume this client exists
        var response = await _client.GetAsync($"clients/{clientId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualClient = JsonConvert.DeserializeObject<Client>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_client_pool().GetClient(clientId), actualClient);

    }

    [Fact]
    public async Task Get_Client_Orders()
    {
        var clientId = 1; // Assume this client exists
        var response = await _client.GetAsync($"clients/{clientId}/orders");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualClients = JsonConvert.DeserializeObject<List<Order>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_order_pool().GetOrdersForClient(clientId), actualClients);
    }

    [Fact]
    public async Task Get_Non_Existent_Client()
    {
        var clientId = -1; // Assume this client does not exist
        var response = await _client.GetAsync($"clients/{clientId}");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }
    #endregion Clients
    #region Shipments

    [Fact]
    public async Task Get_All_Shipments()
    {
        var response = await _client.GetAsync("shipments");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualShipments = JsonConvert.DeserializeObject<List<Shipment>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_shipment_pool().GetShipments(), actualShipments);
    }


    [Fact]
    public async Task Get_Shipment_By_Id()
    {
        var shipmentId = 1; // Assume this shipment exists
        var response = await _client.GetAsync($"shipments/{shipmentId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualShipment = JsonConvert.DeserializeObject<Shipment>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_shipment_pool().GetShipment(shipmentId), actualShipment);


    }

    [Fact]
    public async Task Get_Shipment_Orders()
    {
        var shipmentId = 1; // Assume this shipment exists
        var response = await _client.GetAsync($"shipments/{shipmentId}/orders");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualShipments = JsonConvert.DeserializeObject<List<Shipment>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_shipment_pool().GetShipments(), actualShipments);
    }

    [Fact]
    public async Task Get_Shipment_items()
    {
        var shipmentId = 1; // Assume this shipment exists
        var response = await _client.GetAsync($"shipments/{shipmentId}/orders");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualItems = JsonConvert.DeserializeObject<List<ShipmentItem>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_shipment_pool().GetItemsInShipment(shipmentId), actualItems);
    }


    [Fact]
    public async Task Get_Non_Existent_Shipment()
    {
        var shipmentId = -1; // Assume this shipment does not exist
        var response = await _client.GetAsync($"shipments/{shipmentId}");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }
    #endregion Shipments
    #region ItemGroups

    [Fact]
    public async Task Get_All_ItemGroups()
    {
        var response = await _client.GetAsync("item_groups");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualItemGroups = JsonConvert.DeserializeObject<List<ItemGroup>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_itemgroup_pool().GetItemGroups(), actualItemGroups);
    }


    [Fact]
    public async Task Get_ItemGroup_By_Id()
    {
        var item_groupId = 1; // Assume this item_group exists
        var response = await _client.GetAsync($"item_groups/{item_groupId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualItemGroup = JsonConvert.DeserializeObject<ItemGroup>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_itemgroup_pool().GetItemGroup(item_groupId), actualItemGroup);


    }


    [Fact]
    public async Task Get_ItemGroup_Items()
    {
        var item_groupId = 1; // Assume this item_group exists
        var response = await _client.GetAsync($"item_groups/{item_groupId}/orders");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualItems = JsonConvert.DeserializeObject<List<Item>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_item_pool().GetItemsForItemGroup(item_groupId), actualItems);
    }


    [Fact]
    public async Task Get_Non_Existent_ItemGroup()
    {
        var item_groupId = -1; // Assume this item_group does not exist
        var response = await _client.GetAsync($"item_groups/{item_groupId}");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }
    #endregion ItemGroups
    #region Suppliers

    [Fact]
    public async Task Get_All_Suppliers()
    {
        var response = await _client.GetAsync("suppliers");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualSuppliers = JsonConvert.DeserializeObject<List<Supplier>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_supplier_pool().GetSuppliers(), actualSuppliers);
    }

    [Fact]
    public async Task Get_Supplier_By_Id()
    {
        int supplierId = 1; // Example supplier ID
        var response = await _client.GetAsync($"suppliers/{supplierId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualSupplier = JsonConvert.DeserializeObject<Supplier>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_supplier_pool().GetSupplier(supplierId), actualSupplier);
    }
    
    [Fact]
    public async Task Get_Non_Existent_Supplier()
    {
        int supplierId = -1; // Example non-existent supplier ID
        var response = await _client.GetAsync($"suppliers/{supplierId}");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }
    #endregion Suppliers
    #region Transfers

    [Fact]
    public async Task Get_All_Transfers()
    {
        var response = await _client.GetAsync("transfers");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualTransfers = JsonConvert.DeserializeObject<List<Transfer>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_transfer_pool().GetTransfers(), actualTransfers);
    }

    [Fact]
    public async Task Get_Transfer_By_Id()
    {
        int transferId = 1; // Example transfer ID
        var response = await _client.GetAsync($"transfers/{transferId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualTransfer = JsonConvert.DeserializeObject<Transfer>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_transfer_pool().GetTransfer(transferId), actualTransfer);
    }

    [Fact]
    public async Task Get_Non_Existent_Transfer()
    {
        int transferId = -1; // Example non-existent transfer ID
        var response = await _client.GetAsync($"transfers/{transferId}");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    [Fact]
    public async Task Get_Items_For_Transfers()
    {
        var transferId = 1;
        var response = await _client.GetAsync("transfers/items");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualItems = JsonConvert.DeserializeObject<List<Item>>(await response.Content.ReadAsStringAsync());
        var actualTransferItems = actualItems.Select(item => new TransferItem { /* map properties from item to TransferItem */ }).ToList();
        Assert.Equal(DataProvider.fetch_transfer_pool().GetItemsInTransfer(transferId), actualTransferItems);
    }
    #endregion Transfers
    #region Warehouses

    [Fact]
    public async Task Get_All_Warehouses()
    {
        var response = await _client.GetAsync("warehouses");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualWarehouses = JsonConvert.DeserializeObject<List<Warehouse>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_warehouse_pool().GetWarehouses(), actualWarehouses);
    }

    [Fact]
    public async Task Get_Warehouse_By_Id()
    {
        int warehouseId = 1; // Example warehouse ID
        var response = await _client.GetAsync($"warehouses/{warehouseId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualWarehouse = JsonConvert.DeserializeObject<Warehouse>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_warehouse_pool().GetWarehouse(warehouseId), actualWarehouse);
    }

    [Fact]
    public async Task Get_Non_Existent_Warehouse()
    {
        int warehouseId = -1; // Example non-existent warehouse ID
        var response = await _client.GetAsync($"warehouses/{warehouseId}");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }
    #endregion Warehouses
    #region ItemTypes

    [Fact]
    public async Task Get_All_ItemTypes()
    {
        var response = await _client.GetAsync("item_types");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualItemTypes = JsonConvert.DeserializeObject<List<ItemType>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_itemtype_pool().GetItemTypes(), actualItemTypes);
    
    }

    [Fact]
    public async Task Get_ItemType_By_Id()
    {
        var item_typeId = 1; // Assume this item_type exists
        var response = await _client.GetAsync($"item_types/{item_typeId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualItemType = JsonConvert.DeserializeObject<ItemType>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_itemtype_pool().GetItemType(item_typeId), actualItemType);
    }
    
    [Fact]
    public async Task Get_ItemType_By_Invalid_Id()
    {
        var item_typeId = -1; // Assume this item_type does not exist
        var response = await _client.GetAsync($"item_types/{item_typeId}");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }

    [Fact]
    public async Task Get_Non_Existent_ItemType()
    {
        var item_typeId = -1; // Assume this item_type does not exist
        var response = await _client.GetAsync($"item_types/{item_typeId}");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }
    #endregion ItemTypes
    #region Item

    [Fact]
    public async Task Get_All_Items()
    {
        var response = await _client.GetAsync("items");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualItems = JsonConvert.DeserializeObject<List<Item>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_item_pool().GetItems(), actualItems);
    }

    [Fact]
    public async Task Get_Item_By_Id()
    {
        var itemId = 1; // Assume this item exists
        var response = await _client.GetAsync($"items/{itemId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualItem = JsonConvert.DeserializeObject<Item>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_item_pool().GetItem(itemId.ToString()), actualItem);
    }

    [Fact]
    public async Task Get_Non_Existent_Item()
    {
        var itemId = -1; // Assume this item does not exist
        var response = await _client.GetAsync($"items/{itemId}");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }
    #endregion Item
    #region Orders

    [Fact]
    public async Task Get_All_Orders()
    {
        var response = await _client.GetAsync("orders");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualOrders = JsonConvert.DeserializeObject<List<Order>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_order_pool().GetOrders(), actualOrders);
    }

    [Fact]
    public async Task Get_Order_By_Id()
    {
        var orderId = 1; // Assume this order exists
        var response = await _client.GetAsync($"orders/{orderId}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualOrder = JsonConvert.DeserializeObject<Order>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_order_pool().GetOrder(orderId), actualOrder);
    }

    [Fact]
    public async Task Get_Non_Existent_Order()
    {
        var orderId = -1; // Assume this order does not exist
        var response = await _client.GetAsync($"orders/{orderId}");
        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
    }
    #endregion Orders
    #region Inventories

    // INVENTORIES
    [Fact]
    public async Task Get_All_Inventories() // happy
    {
        var response = await _client.GetAsync("Inventories");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualInventories = JsonConvert.DeserializeObject<List<Inventory>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_inventory_pool().GetInventories(), actualInventories);
    }

    [Fact]
    public async Task Get_One_Inventory() // happy
    {
        var InvID = 1;
        var response = await _client.GetAsync($"Inventories/{InvID}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualInventory = JsonConvert.DeserializeObject<Inventory>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_inventory_pool().GetInventory(InvID), actualInventory);
    }

    [Fact]
    public async Task Get_Inventories_Item() // happy
    {
        var InvID = "P000001";
        var response = await _client.GetAsync($"items/{InvID}/inventory");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualInventories = JsonConvert.DeserializeObject<List<Inventory>>(await response.Content.ReadAsStringAsync());
        var expectedInventories = DataProvider.fetch_inventory_pool().GetInventoriesForItem(InvID);
        Assert.Equal(expectedInventories, actualInventories);
    }

    [Fact]
    public async Task Get_Inventory_Totals_Item() // happy
    {
        var itemId = "P000001"; // Using the provided item ID
        var response = await _client.GetAsync($"items/{itemId}/inventory/totals");

        Assert.Equal(HttpStatusCode.OK, response.StatusCode);

        var actualTotals = JsonConvert.DeserializeObject<Dictionary<string, int>>(await response.Content.ReadAsStringAsync());
        var expectedTotals = DataProvider.fetch_inventory_pool().GetInventoryTotalsForItem(itemId);

        Assert.Equal(expectedTotals["total_expected"], actualTotals["total_expected"]);
        Assert.Equal(expectedTotals["total_ordered"], actualTotals["total_ordered"]);
        Assert.Equal(expectedTotals["total_allocated"], actualTotals["total_allocated"]);
        Assert.Equal(expectedTotals["total_available"], actualTotals["total_available"]);
    }
    #endregion Inventories
    #region Itemlines

    // Item_Lines
    [Fact]
    public async Task Get_All_ItemLines() // happy
    {
        var response = await _client.GetAsync("item_lines");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualLines = JsonConvert.DeserializeObject<List<ItemLine>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_itemline_pool().GetItemLines(), actualLines);
    }

    [Fact]
    public async Task Get_One_ItemLine() // happy
    {
        var ILID = 1;
        var response = await _client.GetAsync($"item_lines/{ILID}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualLines = JsonConvert.DeserializeObject<ItemLine>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_itemline_pool().GetItemLine(ILID), actualLines);
    }
    #endregion Itemlines
    #region Locations

    //Locations
    [Fact]
    public async Task Get_All_Locations() // happy
    {
        var response = await _client.GetAsync("locations");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualLocations = JsonConvert.DeserializeObject<List<Location>>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_location_pool().GetLocations(), actualLocations);
    }

    [Fact]
    public async Task Get_One_Location() // happy
    {
        var LocID = 1;
        var response = await _client.GetAsync($"locations/{LocID}");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
        var actualLocations = JsonConvert.DeserializeObject<Location>(await response.Content.ReadAsStringAsync());
        Assert.Equal(DataProvider.fetch_location_pool().GetLocation(LocID), actualLocations);
    }

    [Fact]
    public async Task Get_Location_With_Warehouse() // happy
    {
        var warehouseId = 8; //use 8 because it only has 168 entries, faster test run
        var response = await _client.GetAsync($"warehouses/{warehouseId}/locations");
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);

        var actualLocations = JsonConvert.DeserializeObject<List<Location>>(await response.Content.ReadAsStringAsync());
        var expectedLocations = DataProvider.fetch_location_pool().GetLocationsInWarehouse(warehouseId);

        Assert.Equal(expectedLocations, actualLocations);
    }
    #endregion Locations

}

File: C#api/Tests/test_get_handler.py
import httpx
import unittest

class ApiGetTests(unittest.TestCase):
    @classmethod
    def setUpClass(self):
        self.base_url = "http://localhost:3000/api/v1/"
        self.client = httpx.Client(headers={"API_KEY": "a1b2c3d4e5"})

    # Clients
    def test_get_all_clients(self):
        response = self.client.get(f"{self.base_url}clients")
        self.assertEqual(response.status_code, 200)

    def test_get_client_by_id(self):
        client_id = 1
        response = self.client.get(f"{self.base_url}clients/{client_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Id'], client_id)

    def test_get_client_orders(self):
        client_id = 1
        response = self.client.get(f"{self.base_url}clients/{client_id}/orders")
        self.assertEqual(response.status_code, 200)

    def test_get_non_existent_client(self):
        client_id = -1
        response = self.client.get(f"{self.base_url}clients/{client_id}")
        self.assertEqual(response.status_code, 404)

    # Shipments
    def test_get_all_shipments(self):
        response = self.client.get(f"{self.base_url}shipments")
        self.assertEqual(response.status_code, 200)

    def test_get_shipment_by_id(self):
        shipment_id = 1
        response = self.client.get(f"{self.base_url}shipments/{shipment_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Id'], shipment_id)

    def test_get_shipment_orders(self):
        shipment_id = 1
        response = self.client.get(f"{self.base_url}shipments/{shipment_id}/orders")
        self.assertEqual(response.status_code, 200)

    def test_get_shipment_items(self):
        shipment_id = 1
        response = self.client.get(f"{self.base_url}shipments/{shipment_id}/orders")
        self.assertEqual(response.status_code, 200)

    def test_get_non_existent_shipment(self):
        shipment_id = -1
        response = self.client.get(f"{self.base_url}shipments/{shipment_id}")
        self.assertEqual(response.status_code, 404)

    # ItemGroups
    def test_get_all_item_groups(self):
        response = self.client.get(f"{self.base_url}item_groups")
        self.assertEqual(response.status_code, 200)

    def test_get_item_group_by_id(self):
        item_group_id = 1
        response = self.client.get(f"{self.base_url}item_groups/{item_group_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Id'], item_group_id)

    def test_get_item_group_items(self):
        item_group_id = 1
        response = self.client.get(f"{self.base_url}item_groups/{item_group_id}/items")
        self.assertEqual(response.status_code, 200)

    def test_get_non_existent_item_group(self):
        item_group_id = -1
        response = self.client.get(f"{self.base_url}item_groups/{item_group_id}")
        self.assertEqual(response.status_code, 404)

    # Suppliers
    def test_get_all_suppliers(self):
        response = self.client.get(f"{self.base_url}suppliers")
        self.assertEqual(response.status_code, 200)

    def test_get_supplier_by_id(self):
        supplier_id = 1
        response = self.client.get(f"{self.base_url}suppliers/{supplier_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Id'], supplier_id)

    def test_get_supplier_items(self):
        supplier_id = 1
        response = self.client.get(f"{self.base_url}suppliers/{supplier_id}/items")
        self.assertEqual(response.status_code, 200)

    def test_get_non_existent_supplier(self):
        supplier_id = -1
        response = self.client.get(f"{self.base_url}suppliers/{supplier_id}")
        self.assertEqual(response.status_code, 404)

    # Transfers
    def test_get_all_transfers(self):
        response = self.client.get(f"{self.base_url}transfers")
        self.assertEqual(response.status_code, 200)

    def test_get_transfer_by_id(self):
        transfer_id = 1
        response = self.client.get(f"{self.base_url}transfers/{transfer_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Id'], transfer_id)

    def test_get_non_existent_transfer(self):
        transfer_id = -1
        response = self.client.get(f"{self.base_url}transfers/{transfer_id}")
        self.assertEqual(response.status_code, 404)

    def test_get_items_for_transfers(self):
        transfer_id = 1
        response = self.client.get(f"{self.base_url}transfers/{transfer_id}/items")
        self.assertEqual(response.status_code, 200)

    # Warehouses
    def test_get_all_warehouses(self):
        response = self.client.get(f"{self.base_url}warehouses")
        self.assertEqual(response.status_code, 200)

    def test_get_warehouse_by_id(self):
        warehouse_id = 1
        response = self.client.get(f"{self.base_url}warehouses/{warehouse_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Id'], warehouse_id)

    def test_get_non_existent_warehouse(self):
        warehouse_id = -1
        response = self.client.get(f"{self.base_url}warehouses/{warehouse_id}")
        self.assertEqual(response.status_code, 404)

    # ItemTypes
    def test_get_all_item_types(self):
        response = self.client.get(f"{self.base_url}item_types")
        self.assertEqual(response.status_code, 200)

    def test_get_item_type_by_id(self):
        item_type_id = 1
        response = self.client.get(f"{self.base_url}item_types/{item_type_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Id'], item_type_id)

    def test_get_item_type_items(self):
        item_type_id = 1
        response = self.client.get(f"{self.base_url}item_types/{item_type_id}/items")
        self.assertEqual(response.status_code, 200)

    def test_get_non_existent_item_type(self):
        item_type_id = -1
        response = self.client.get(f"{self.base_url}item_types/{item_type_id}")
        self.assertEqual(response.status_code, 404)

    # Items
    def test_get_all_items(self):
        response = self.client.get(f"{self.base_url}items")
        self.assertEqual(response.status_code, 200)

    def test_get_item_by_id(self):
        item_id = "P000047"
        response = self.client.get(f"{self.base_url}items/{item_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Uid'], item_id)

    def test_get_non_existent_item(self):
        item_id = "1"
        response = self.client.get(f"{self.base_url}items/{item_id}")
        self.assertEqual(response.status_code, 404)

    # Orders
    def test_get_all_orders(self):
        response = self.client.get(f"{self.base_url}orders")
        self.assertEqual(response.status_code, 200)

    def test_get_order_by_id(self):
        order_id = 1
        response = self.client.get(f"{self.base_url}orders/{order_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Id'], order_id)

    def test_get_non_existent_order(self):
        order_id = -1
        response = self.client.get(f"{self.base_url}orders/{order_id}")
        self.assertEqual(response.status_code, 404)

    # Inventories
    def test_get_all_inventories(self):
        response = self.client.get(f"{self.base_url}inventories")
        self.assertEqual(response.status_code, 200)

    def test_get_inventory_by_id(self):
        inventory_id = 1
        response = self.client.get(f"{self.base_url}inventories/{inventory_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Id'], inventory_id)

    def test_get_non_existent_inventory(self):
        inventory_id = -1
        response = self.client.get(f"{self.base_url}inventories/{inventory_id}")
        self.assertEqual(response.status_code, 404)

    # ItemLines
    def test_get_all_item_lines(self):
        response = self.client.get(f"{self.base_url}item_lines")
        self.assertEqual(response.status_code, 200)

    def test_get_item_line_by_id(self):
        item_line_id = 1
        response = self.client.get(f"{self.base_url}item_lines/{item_line_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Id'], item_line_id)

    def test_get_non_existent_item_line(self):
        item_line_id = -1
        response = self.client.get(f"{self.base_url}item_lines/{item_line_id}")
        self.assertEqual(response.status_code, 404)

    # Locations
    def test_get_all_locations(self):
        response = self.client.get(f"{self.base_url}locations")
        self.assertEqual(response.status_code, 200)

    def test_get_location_by_id(self):
        location_id = 1
        response = self.client.get(f"{self.base_url}locations/{location_id}")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.json()['Id'], location_id)

    def test_get_non_existent_location(self):
        location_id = -1
        response = self.client.get(f"{self.base_url}locations/{location_id}")
        self.assertEqual(response.status_code, 404)

if __name__ == '__main__':
    unittest.main()
